Blog: A Non-coders Thoughts on Everyone Codes - Part Two (#3476)
PostHog/posthog.com: contents/blog/a-non-coders-thoughts-on-everybody-codes-culture-part-two.md

---
date: 2022-05-27
title: A non-coders thoughts on ‘Everybody Codes’ - Part Two
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
categories: ["Inside PostHog"]

> featuredImage: ../images/blog/posthog-blog-image.png

Reviewer: ```suggestion
featuredImage: ../images/blog/migrating-hog.png
```

featuredImageType: full
---

> One of PostHog’s values is that everybody codes. That doesn’t mean everyone needs to bea developer, but we do encourage everyone to practice the basics of shipping, no matter their role. Even site updates require GitHub and  some basic knowledge of coding, for example. 

Reviewer: ```suggestion
One of PostHog’s values is that everybody codes. That doesn’t mean everyone needs to be a developer, but we do encourage everyone to practice the basics of shipping, no matter their role. Even site updates require GitHub and  some basic knowledge of coding, for example. 
```

If I’d stuck to realistic limits, that would be the end of it. I can’t  build a product feature on my own, can I?

It turns out that I can. Late last year I made [a ShuffleHog prototype on Twitter](https://twitter.com/thehogbot), coded in [Tracery](https://www.tracery.io/). That lasted until [Marius added the ability to make front-end apps in PostHog](https://github.com/PostHog/posthog/issues/9654#issuecomment-1133222836) last week. 24 hours later, ShuffleHog now lives inside PostHog as a beta feature that can suggest random query ideas.

![ShuffleHog - AKA PostHog's Best Feature](../images/blog/non-coders-thoughts/shufflehog_app.png)

ShuffleHog still isn’t quite ready for other users to try yet, but who knows? Maybe I’ll keep following this dream. 

### (Keep) making mistakes
One thing that hasn’t changed over the last 12 months is the importance of making mistakes — and I should know. I make a lot of them.

The reason mistakes are so important is, I think, because coding isn’t a prescriptive discipline. There’s no single correct way to code or write, so it’s more valuable to know why something _broke_ than how it ‘should’ be done. 

> Breaking things is a great way to learn about how to fix them.PostHog recently rebranded our plugins to apps and, as part of that, we launched a new apps library.  I broke a bunch of things while working on that project — but that's OK because I managed to fix most of them (with occasional help from Eli) and carried those lessons over to the ShuffleHog project.

Reviewer: ```suggestion
Breaking things is a great way to learn about how to fix them. PostHog recently rebranded plugins to apps and, as part of that, we launched a new apps library. I broke a bunch of things while working on that project — but that's OK because I managed to fix most of them (with occasional help from Eli) and carried those lessons over to the ShuffleHog project.
```

### Take it offline
Something I’ve loved about becoming more confident with code is that I’ve been able to follow this passion outside of work. This creates a feedback loop where I get even more excited about what I’m learning and what I’m able to do, so I bring that energy back into work…where I _also_ learn and do more. It’s very satisfying. 

![Joe's Open Source Arcade](../images/blog/non-coders-thoughts/arcade_open_source.png)

> The experience I’ve had with GitHub, for example, ended up being the catalyst for me to build my own arcade emulation machine at home because I now had the skills to fork and tweak certain repos. Likewise, spending time in the world of open source software led to me playing around with a _lot_ of Raspberry Pi sensors at home.

Reviewer: ```suggestion
The experience I’ve had with GitHub, for example, ended up being the catalyst for me to build my own arcade emulation machine at home because I now had the skills to fork and tweak certain repos. Likewise, spending time in the world of open-source software led to me playing around with a _lot_ of Raspberry Pi sensors at home.
```
